hazelcast/src/test/java/com/hazelcast/cache/merge/CacheSplitBrainTest.java

```diff
@@ -0,0 +1,380 @@
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.NightlyTest;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import javax.cache.Cache;
+import javax.cache.CacheManager;
+import javax.cache.spi.CachingProvider;
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(NightlyTest.class)
+public class CacheSplitBrainTest extends HazelcastTestSupport {
+
+    @Before
+    @After
+    public void killAllHazelcastInstances() {
+        HazelcastInstanceFactory.terminateAll();
+    }
+
+    @Test
+    public void testLatestAccessCacheMergePolicy() {
+        String cacheName = randomMapName();
+        Config config = newConfig();
+        HazelcastInstance h1 = Hazelcast.newHazelcastInstance(config);
+        HazelcastInstance h2 = Hazelcast.newHazelcastInstance(config);
+
+        TestMemberShipListener memberShipListener = new TestMemberShipListener(1);
+        h2.getCluster().addMembershipListener(memberShipListener);
+        TestLifeCycleListener lifeCycleListener = new TestLifeCycleListener(1);
+        h2.getLifecycleService().addLifecycleListener(lifeCycleListener);
+
+        closeConnectionBetween(h1, h2);
+
+        assertOpenEventually(memberShipListener.latch);
+        assertClusterSizeEventually(1, h1);
+        assertClusterSizeEventually(1, h2);
+
+        CachingProvider cachingProvider1 = HazelcastServerCachingProvider.createCachingProvider(h1);
+        CachingProvider cachingProvider2 = HazelcastServerCachingProvider.createCachingProvider(h2);
+
+        CacheManager cacheManager1 = cachingProvider1.getCacheManager();
+        CacheManager cacheManager2 = cachingProvider2.getCacheManager();
+
+        CacheConfig cacheConfig = newCacheConfig(cacheName, LatestAccessCacheMergePolicy.class.getName());
+
+        Cache cache1 = cacheManager1.createCache(cacheName, cacheConfig);
+        Cache cache2 = cacheManager2.createCache(cacheName, cacheConfig);
+
+        // TODO We assume that until here and also while doing get/put, cluster is still splitted.
+        // This assumptions seems fragile due to time sensitivity.
+
+        cache1.put("key1", "value");
+        cache1.get("key1"); // Access to record
+
+        // Prevent updating at the same time
+        sleepAtLeastMillis(1);
+
+        cache2.put("key1", "LatestUpdatedValue");
+        cache2.get("key1"); // Access to record
+
+        cache2.put("key2", "value2");
+        cache2.get("key1"); // Access to record
+
+        // Prevent updating at the same time
+        sleepAtLeastMillis(1);
+
+        cache1.put("key2", "LatestUpdatedValue2");
+        cache1.get("key2"); // Access to record
```

Comment:
can we make here as `cache2.get("key2");` then ` assertEquals("value2", cacheTest.get("key2"));`
because policy is latestAccess, and it is nice to see `get` operation effect.